pkg/idtools/idtools.go

```diff
@@ -38,13 +38,20 @@ const (
 // ownership to the requested uid/gid.  If the directory already exists, this
 // function will still change ownership to the requested uid/gid pair.
 func MkdirAllAs(path string, mode os.FileMode, ownerUID, ownerGID int) error {
```

Comment:
The intent of the function is not changed, but this PR fixes a bug (or at least an assumed bug) you mentioned in a comment that got lost in the larger stream of comments in the userns PR: now the "as uid/gid" (chown) part of the function happens to all components of the path, not just the topmost dir.  The new function (`MkdirAllNewAs`) skips chown of existing directories.